src/Abstraction/Result.php

```diff
@@ -2,15 +2,17 @@
 
 declare(strict_types=1);
 
-namespace Doctrine\DBAL;
+namespace Doctrine\DBAL\Abstraction;
 
-use Doctrine\DBAL\Driver\ResultStatement as DriverResultStatement;
+use Doctrine\DBAL\DBALException;
+use Doctrine\DBAL\Driver\Result as DriverResult;
 use Traversable;
 
 /**
- * DBAL-level ResultStatement interface.
+ * Abstraction-level result statement execution result. Provides additional methods on top
+ * of the driver-level interface.
  */
-interface ResultStatement extends DriverResultStatement
+interface Result extends DriverResult
```

Comment:
> That interface is not meant to grow, is it?

I believe it will grow. This is the interface that the DBAL-level result will implement and it's not limited by iteration. E.g. it may implement new `fetch*()` or `fetchAll*()` methods if we decide so.